server/src/utils/mixerConnections/AtemConnection.ts

```diff
@@ -0,0 +1,415 @@
+                    this._connection.setFairlightAudioMixerSourceProps(
+                        this._chNoToSource[channelIndex],
+                        '-255',
+                        {
+                            mixOption: FairlightAudioMixOption.On,
+                        }
+                    )
+                }
+
+                this._sourceTracks[channelTypeIndex] = '-255'
+                break
+        }
+    }
+
+    updateFadeIOLevel(channelIndex: number, level: number): void {
+        const channel = this.getChannel(channelIndex)
+        const fader = this.getFaderFromChannelIndex(channelIndex)
+        const { muteOn } = this.getFaderFromChannelIndex(channelIndex)
+        const sourceNo = this._chNoToSource[channel.channelTypeIndex]
+
+        if (level > 0 && !muteOn) {
+            this._connection.setFairlightAudioMixerSourceProps(
+                sourceNo,
+                this._sourceTracks[channel.channelTypeIndex],
+                { mixOption: FairlightAudioMixOption.On }
+            )
+            this._connection.setFairlightAudioMixerSourceProps(
+                sourceNo,
+                this._sourceTracks[channel.channelTypeIndex],
+                {
+                    faderGain: Math.round(
+                        Math.max(-10000, Math.min(1000, floatToDB(level) * 100))
+                    ),
+                }
+            )
+        } else {
+            this._connection.setFairlightAudioMixerSourceProps(
+                sourceNo,
+                this._sourceTracks[channel.channelTypeIndex],
+                { mixOption: FairlightAudioMixOption.Off }
+            )
+            this._connection.setFairlightAudioMixerSourceProps(
+                sourceNo,
+                this._sourceTracks[channel.channelTypeIndex],
+                {
+                    faderGain: 0,
+                }
+            )
+        }
+    }
+
+    private getFaderFromChannelIndex(channelIndex: number): IFader {
+        const faderIndex =
+            state.channels[0].chMixerConnection[this.mixerIndex].channel[
+                channelIndex
+            ].assignedFader
+
+        return state.faders[0].fader[faderIndex]
+    }
+    private getChannel(channelIndex: number): IChannel {
+        return state.channels[0].chMixerConnection[this.mixerIndex].channel[
+            channelIndex
+        ]
+    }
+
+    private setPropsFromChannel(
+        channelIndex: number,
+        channel: IChannel,
+        fader: IFader,
+        source: FairlightAudioSource
+    ) {
+        if ((source.properties.gain + 1200) / 1800 !== fader.inputGain) {
+            // update gain
+            this.setFaderGain(
+                channelIndex,
+                (source.properties.gain + 1200) / 1800
+            )
+        }
+        if (!channel.fadeActive) {
+            if (
+                source.properties.mixOption === FairlightAudioMixOption.On &&
+                source.properties.faderGain > -10000
```

Comment:
What does this condition check for?

Reply:
Whether any audio is being let through, i.e. the fader needs to up and the channel needs to be on.